src/components/LanguagePicker/index.tsx

```diff
@@ -0,0 +1,226 @@
+import {
+  Box,
+  Flex,
+  FormControl,
+  FormLabel,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Kbd,
+  Menu,
+  MenuList,
+  type MenuListProps,
+  type MenuProps,
+  Text,
+  type UseDisclosureReturn,
+  useEventListener,
+} from "@chakra-ui/react"
+
+import { Button } from "@/components/Buttons"
+import { BaseLink } from "@/components/Link"
+
+import MenuItem from "./MenuItem"
+import NoResultsCallout from "./NoResultsCallout"
+import { useLanguagePicker } from "./useLanguagePicker"
+
+type LanguagePickerProps = Omit<MenuListProps, "children"> & {
+  children: React.ReactNode
+  placement?: MenuProps["placement"]
+  handleClose?: () => void
+  menuState?: UseDisclosureReturn
+}
+
+const LanguagePicker = ({
+  children,
+  placement,
+  handleClose,
+  menuState,
+  ...props
+}: LanguagePickerProps) => {
+  const { t, refs, disclosure, filterValue, setFilterValue, filteredNames } =
+    useLanguagePicker(handleClose, menuState)
+  const { inputRef, firstItemRef, noResultsRef, footerRef } = refs
+  const { onClose } = disclosure
+
+  /**
+   * Adds a keydown event listener to focus filter input (\).
+   * @param {string} event - The keydown event.
+   */
+  useEventListener("keydown", (e) => {
+    if (e.key !== "\\") return
+    e.preventDefault()
+    inputRef.current?.focus()
+  })
+
+  return (
+    <Menu isLazy placement={placement} autoSelect={false} {...disclosure}>
+      {children}
+      <MenuList
+        position="relative"
+        overflow="auto"
+        borderRadius="base"
+        py="0"
+        onKeyDown={(e) => {
+          if (e.key === "Tab" || e.key === "\\") {
+            e.preventDefault()
+            ;(e.shiftKey ? inputRef : footerRef).current?.focus()
```

Comment:
This is cool, but I have a feeling that most users will never notice that it exists. For future iterations, we could add a help icon/tooltip or a shortcuts summary that explains this.